app/models/manageiq/providers/vmware/infra_manager/provision_workflow.rb

```diff
@@ -144,6 +144,16 @@ def available_vlans_and_hosts(options = {})
     return vlans, hosts
   end
 
+  def filter_hosts_by_vlan_name(all_hosts)
+    vlan_name = get_value(@values[:vlan])
+    return all_hosts unless vlan_name
+
+    _log.info("Filtering hosts with the following network: <#{vlan_name}>")
+    shared = !vlan_name.match(/dvs_/).nil?
+    vlan_name.sub!(/^dvs_/, '') if shared
+    all_hosts.select { |h| h.lans.any? { |lan| lan.name == vlan_name && lan.switch.shared == shared } }
```

Comment:
Not really "clean" but you could do `&& !!lan.switch.shared == shared` so a `nil` is treated as false